internal/handlers/pg/pgdb/databases.go

```diff
@@ -58,9 +58,20 @@ func Databases(ctx context.Context, querier pgxtype.Querier) ([]string, error) {
 
 // CreateDatabase creates a new FerretDB database (PostgreSQL schema).
 //
-// It returns (possibly wrapped) ErrAlreadyExist if schema already exist,
-// use errors.Is to check the error.
+// It returns (possibly wrapped):
+//
+//   * ErrAlreadyExist if schema already exist.
+//   * ErrInvalidDatabaseName if db name doesn't comply with the rules.
+// Use errors.Is to check the error.
 func CreateDatabase(ctx context.Context, querier pgxtype.Querier, db string) error {
+	if !validateDatabaseNameRe.MatchString(db) {
+		return ErrInvalidDatabaseName
+	}
+
+	if strings.HasPrefix(db, reservedPrefix) {
+		return ErrInvalidDatabaseName
+	}
```

Comment:
As it's the same error returned, why not both conditions together?

```suggestion
	if !validateDatabaseNameRe.MatchString(db) || strings.HasPrefix(db, reservedPrefix) {
		return ErrInvalidDatabaseName
	}
```

Reply:
> As it's the same error returned, why not both conditions together?

Well, it can be, but it's polishing.